strpipe/ops/pad.pyx

```diff
@@ -12,43 +13,149 @@ from strpipe.data.types import STRING_LIST
 
 
 cdef class Pad(BaseOp):
+    '''Pad texts so that they are the same size.
+
+    Args:
+        pad_token (optional): Padding token. Default: :obj:`DefaultTokens.pad`
+        sos_token (optional): Start of sentence token.
+        eos_token (optional): End of sentence token.
+        maxlen (optional): The length to pad to. Note that this includes eos, sos tokens as
+            part of the length count.
+    '''
 
     cdef str _pad_token
+    cdef str _sos_token
+    cdef str _eos_token
     cdef int _maxlen
 
-    def __init__(self, str pad_token=DefaultTokens.pad, int maxlen=-1):
+    def __init__(
+        self,
+        str sos_token=DefaultTokens.nul,
+        str eos_token=DefaultTokens.nul,
+        str pad_token=DefaultTokens.pad,
+        int maxlen=-1):
         self.input_type = STRING_LIST
         self.output_type = STRING_LIST
         self._pad_token = pad_token
+
+        if (sos_token == DefaultTokens.nul) != (eos_token == DefaultTokens.nul):
```

Comment:
Maybe wrap `token == DefaultTokens.nul` into a function `cstr_is_none`? I spent a lot of time to figureout what it does.

Reply:
:ok_hand: 
Added `_check_tokens`